src/Symfony/Component/Notifier/Bridge/RocketChat/RocketChatTransport.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\RocketChat;
+
+use Symfony\Component\Notifier\Exception\LogicException;
+use Symfony\Component\Notifier\Exception\TransportException;
+use Symfony\Component\Notifier\Message\ChatMessage;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Jeroen Spee <spee.jeroen@gmail.com>
+ *
+ * @internal
+ *
+ * @experimental in 5.0
+ */
+final class RocketChatTransport extends AbstractTransport
+{
+    protected const HOST = 'rocketchat.com';
+
+    private $accessToken;
+    private $chatChannel;
+
+    public function __construct(
+        string $accessToken,
+        string $chatChannel = null,
+        HttpClientInterface $client = null,
+        EventDispatcherInterface $dispatcher = null
+    ) {
+        $this->accessToken = $accessToken;
+        $this->chatChannel = $chatChannel;
+        $this->client = $client;
+
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('rocketchat://%s?channel=%s', $this->getEndpoint(), $this->chatChannel);
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof ChatMessage && (null === $message->getOptions() || $message->getOptions() instanceof RocketChatOptions);
+    }
+
+    /**
+     * @see https://rocket.chat/docs/administrator-guides/integrations/
+     */
+    protected function doSend(MessageInterface $message): void
+    {
+        if (!$message instanceof ChatMessage) {
+            throw new LogicException(sprintf('The "%s" transport only supports instances of "%s" (instance of "%s" given).', __CLASS__, ChatMessage::class, \get_class($message)));
+        }
+        if ($message->getOptions() && !$message->getOptions() instanceof RocketChatOptions) {
+            throw new LogicException(sprintf('The "%s" transport only supports instances of "%s" for options.', __CLASS__, RocketChatOptions::class));
+        }
+
+        $options = $message->getOptions()->toArray();
```

Comment:
Hi ! This line throw "Call to a member function toArray() on null" if you try to send message without options.

```
$message = new ChatMessage('Symfony TEST !');
```

Maybe you can test if `$message->getOptions()` is null ?

I tested with the following code but it may be possible to do better.

```
$options = $message->getOptions() instanceof MessageOptionsInterface ? $message->getOptions()->toArray() : [];
```

Reply:
What about: `$options = ($opts = $message->getOptions()) ? $opts->toArray() : [];`
The same code is used in the telegram bridge

Reply:
Thanks, I've changed it to that.